src/main/java/com/example/music_player/storage/CloudStorageAmazonS3.java

```diff
@@ -29,46 +29,50 @@
     @Autowired
     private AmazonS3 s3Client;
     private File tempFile;
-    private File fileObject;
 
     @Override
     public List<Source> save(InputStream inputStream, String originalFilename, String contentType) {
-        fileObject = putInputStreamToFile(inputStream);
-        boolean found = s3Client.doesBucketExistV2(bucketName);
-
-        if (!found) {
-            log.info("IN save() :Bucket " + bucketName + " start creating.");
-            s3Client.createBucket(bucketName);
-        } else {
-            log.info("IN save() :Bucket " + bucketName + "  already exists.");
+        List<Source> sourceList = null;
+        try {
+            putInputStreamToFile(inputStream);
+            if (!s3Client.doesBucketExistV2(bucketName)) {
+                log.info("IN save() :Bucket " + bucketName + " start creating.");
+                s3Client.createBucket(bucketName);
+            } else {
+                log.info("IN save() :Bucket " + bucketName + "  already exists.");
+            }
+            s3Client.putObject(new PutObjectRequest(bucketName, originalFilename, tempFile));
+            sourceList = createSource(originalFilename, contentType);
+        } catch (Exception e) {
+            System.out.println("someException " +e.getMessage());
+        } finally {
+            tempFile.delete();
+            tempFile.deleteOnExit();
```

Comment:
Useless or even harmful method that should be carefully used in specific cases or better should be avoid.